connectors/protocol/connectors.py

```diff
@@ -784,11 +784,17 @@ async def prepare(self, config, sources):
             doc["service_type"] = configured_service_type
             self.log_debug(f"Populated service type {configured_service_type}")
 
+        simple_config = source_klass.get_simple_configuration()
         if self.configuration.is_empty():
             # sets the defaults and the flag to NEEDS_CONFIGURATION
-            doc["configuration"] = source_klass.get_simple_configuration()
+            doc["configuration"] = simple_config
             doc["status"] = Status.NEEDS_CONFIGURATION.value
             self.log_debug("Populated configuration")
+        elif simple_config.keys() - self.configuration.to_dict().keys():
+            missing_keys = simple_config.keys() - self.configuration.to_dict().keys()
+            self.log_warning(f"Detected an existing connector: {self.id} ({self.service_type}) that was previously {Status.CONNECTED.value} but is now missing configuration: {missing_keys}. Values for the new fields will be automatically set. Please review these configuration values as part of your upgrade.")
```

Comment:
How do you intend to get a default value for a new field? I recently did work to [remove not-useful default values from RCFs](https://github.com/elastic/connectors-python/pull/761), so they aren't typicallly available unless the field is a boolean.

Reply:
well in this case, these are booleans and one `int` where I added a default `value`.

We shouldn't be adding new RCFs that need non-trivial values, or that have initial values that will cause a behavior change from the previous version. Such RCFs would need a "real" migration, I think. In a discussion with @sphilipse and @danajuratoni , we felt that we still need a more comprehensive plan for migrations in the future, but that this small subset of "migrations" are something we could whip up pretty quickly and without a lot of risk.

Reply:
And I just removed the depth (int) config, so we're back to just booleans

Reply:
Thinking over it, it's not _such_ a problem if there is no `value`. This migrator will add the new RCF with a type-safe empty value. The configuration won't be flagged as invalid, but if the field is required the validator will flag that the new field is empty and inform the user.

Reply:
I do agree that migrations need to be worked out in the future, but with this change we can effectively migrate already, it's just unclear to the user that a migration happened.
In any case, my initial concern has been abated.